File: HanYeojin/Assignment2/yeojin/cafe-server/src/main/java/com/yeojin/cafe/repository/MenuRepository.java
package com.yeojin.cafe.repository;

import com.yeojin.cafe.domain.Menu;
import org.springframework.stereotype.Repository;

import java.math.BigDecimal;
import java.util.*;
import java.util.concurrent.atomic.AtomicLong;

@Repository
public class MenuRepository {
    private final Map<Long, Menu> menuStore = new HashMap<>();
    private final AtomicLong sequence = new AtomicLong(1);

    // 생성자에서 초기 메뉴 데이터 추가
    public MenuRepository() {
        Menu menu1 = new Menu(sequence.getAndIncrement(), "아메리카노", new BigDecimal("4500"));
        Menu menu2 = new Menu(sequence.getAndIncrement(), "바닐라라떼", new BigDecimal("5500"));
        Menu menu3 = new Menu(sequence.getAndIncrement(), "딸기스무디", new BigDecimal("6000"));
        Menu menu4 = new Menu(sequence.getAndIncrement(), "망고스무디", new BigDecimal("6000"));

        menuStore.put(menu1.getId(), menu1);
        menuStore.put(menu2.getId(), menu2);
        menuStore.put(menu3.getId(), menu3);
        menuStore.put(menu4.getId(), menu4);
    }

    // 전체 메뉴 조회
    public List<Menu> findAllMenu() {
        return new ArrayList<>(menuStore.values());
    }

    // menu ID로 조회
    public Optional<Menu> findMenuById(Long id) {
        return Optional.ofNullable(menuStore.get(id));
    }
}

File: assignment2/MinSeungho/cafe/src/main/kotlin/com/seungho/cafe/repository/MenuRepository.kt
package com.seungho.cafe.repository

import com.seungho.cafe.domain.Menu
import jakarta.annotation.PostConstruct
import org.springframework.stereotype.Repository
import java.util.concurrent.ConcurrentHashMap
import java.util.concurrent.atomic.AtomicLong
import kotlin.collections.set

@Repository
class MenuRepository {
    private val store = ConcurrentHashMap<Long, Menu>()
    private val sequence = AtomicLong(0L)

    @PostConstruct // 메뉴 초기 등록
    fun init() {
        save(Menu(id = -1L, name = "아메리카노", price = 2000L)) // 임시 ID
        save(Menu(id = -1L, name = "카페라떼", price = 3500L))
        save(Menu(id = -1L, name = "유자차", price = 3000L))
    }

    fun save(menu: Menu): Menu {
        val newId = sequence.incrementAndGet()
        val newMenu = menu.copy(id = newId)
        store[newId] = newMenu
        return newMenu
    }

    fun findById(id: Long): Menu? {
        return store[id]
    }

    fun findAll(): List<Menu> {
        return store.values.toList()
    }
}